### ISSUE TITLE
inference error

### ISSUE THREAD

**Issue by INGnowait** [zju3dv/clean-pvnet#207]
您好，在/lib/networks/pvnet/resnet18.py的代码中有下面一个函数：
`    def decode_keypoint(self, output):
        vertex = output['vertex'].permute(0, 2, 3, 1)
        b, h, w, vn_2 = vertex.shape
        vertex = vertex.view(b, h, w, vn_2//2, 2)
        mask = torch.argmax(output['seg'], 1)
        # plt.imshow(torch.argmax(output['seg'], 1).cpu().numpy()[0, :, :])
        # plt.show()
        if cfg.test.un_pnp:
            mean = ransac_voting_layer_v3(mask, vertex, 512, inlier_thresh=0.99)
            kpt_2d, var = estimate_voting_distribution_with_mean(mask, vertex, mean)
            output.update({'mask': mask, 'kpt_2d': kpt_2d, 'var': var})
        else:
            kpt_2d = ransac_voting_layer_v3(mask, vertex, 128, inlier_thresh=0.99, max_num=100)
            output.update({'mask': mask, 'kpt_2d': kpt_2d})`
output是网络的直接输出，不是很明白output['seg']为什么是双通道，mask = torch.argmax(output['seg'], 1)这串代码为什么能生成mask呢？而我得到的mask却为空白的，训练过程是没有问题的。我查看了一下loss的计算方法seg_loss =nn.CrossEntropyLoss()output['seg'], mask)。总的来说就是没明白双通道如何生成mask的，能解答一下吗？谢谢

**Comment by pengsida** [comment]
`mask = torch.argmax(output['seg'], 1)`把双通道的seg变为单通道。
